src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.service.OrderService;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService cartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setOrderDate(LocalDateTime.now());
+        order.setUser(shoppingCart.getUser());
+        order.setTickets(shoppingCart.getTickets());
```

Comment:
When passing tickets from shoppingCart to order - pass the copy of this list(you can use ArrayList constructor for this); The reason is that Hibernate will automatically delete these tickets from shoppingCart(in DB) after noticing them in our order object. Let's do this manually using clear() method from ShoppingCartService after the order with tickets has been created.
```suggestion
        order.setTickets(new ArrayList<>(shoppingCart.getTickets()));
```